diff --git a/arch/arm/mach-rockchip/decode_bl31.py b/arch/arm/mach-rockchip/decode_bl31.py
index 301bd15..d99f7da 100755
--- a/arch/arm/mach-rockchip/decode_bl31.py
+++ b/arch/arm/mach-rockchip/decode_bl31.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 #
 # Copyright (C) 2020 Rockchip Electronics Co., Ltd
 #
@@ -41,7 +41,7 @@ def generate_atf_binary(bl31_file_name):
             atf.write(data)
 
 def main():
-    bl31_elf="./bl31.elf"
+    bl31_elf=os.environ.get("BL31", "./bl31.elf")
     generate_atf_binary(bl31_elf);
 
 if __name__ == "__main__":
diff --git a/arch/arm/mach-rockchip/fit_nodes.sh b/arch/arm/mach-rockchip/fit_nodes.sh
index 8ed2633..bea88cf 100755
--- a/arch/arm/mach-rockchip/fit_nodes.sh
+++ b/arch/arm/mach-rockchip/fit_nodes.sh
@@ -192,6 +192,9 @@ function gen_bl32_node()
 	fi
 
 	TEE="tee.bin"
+	if [ ! -f "${TEE}" ]; then
+		return
+	fi
 	echo "		optee {
 			description = \"OP-TEE\";
 			data = /incbin/(\"${TEE}${SUFFIX}\");
